### DISCUSSION TITLE
Improve full screen experience for desktop apps

### DISCUSSION THREAD

**Discussion by jaigak** [microsoft/WindowsAppSDK#1903] [General]
In most desktop apps (including WinUI 3 apps), the full-screen experience is rather poor even if implemented correctly compare to Universal Windows Platform apps. Issues include:

- No taskbar, title bar Edge UI (See #786). Important for touch users but also useful for mouse/keyboard users. Some video games may not want Edge UI so you could provide a property to disable it under FullScreenPresenter.
- Broken animation when opening the taskbar through the Windows + T keyboard shortcut (Shell issue).
- App window stays behind when you switch to another app or the Windows 10 Start screen. This is different from UWP apps where switching minimizes the full-screen window so the content isn't visible.
- Unlike UWP apps, when you click on the full-screen window icon in the taskbar, it doesn't minimize the window which is inconsistent with standard windows.

Another bug that I noticed with the AppSDK windowing APIs is that, if you full screen a maximized window, the taskbar isn't covered.

**Comment by sylveon** [comment]
> App window stays behind when you switch to another app or the Windows 10 Start screen. This is different from UWP apps where switching minimizes the full-screen window so the content isn't visible.

I don't like the UWP behavior here. With the existing Win32 behavior, I can put a window over it to do other things while matchmaking or loading progresses, and then easily notice when it's done. With UWP, I have to frequently restore the window to check on the status.

**Reply by terrycox** [reply]
Take a look WinUIEx:
https://github.com/dotMorten/WinUIEx

**Reply by jaigak** [reply]
In my opinion, the Win32 behaviour can be sometimes confusing, especially when the new visible window is maximized. Also, it prevents the app from getting suspended so resources will be wasted if not needed (not a problem if the app explicitly disables suspend eg: you may not want games to get suspended while loading because then it will just take longer to load).

**Reply by sylveon** [reply]
I'm not sure how WinUIEx is relevant

**Comment by rkarman** [comment]
Let's start with the feature suggestions first:

**First bullet** - we have this functionality (allowing for taskbar/titlebar to show on edge gesture/interaction while in FullScreen) in the backlog, and though it will not make it for v1.1 I'm hopeful for v1.2. 

**Third bullet** - this is "by design" for how the win32 windowing model works, which is what Windows App SDK builds on top of. We could explore options to allow the app to select which behavior it wants when the window gets occluded by another and loses foreground/input. I'd be interested to see if there are more developers/users who wants this behavior, but we can always add it to ProductBoard and collect more data for whether to prioritize it.

@andrewleader , let's sync up on how to make these two items visible on ProductBoard (note to followers of this topic that want to upvote/add comments in ProductBoard: this will likely happen after the holidays/January).

Then the bugs, as I believe they are based on your description of the behaviors:
**Second bullet** - can you please file a FeedbackHub issue for this if you haven't already? Also, feel free to forward the FeedbackHub issue link to me and I can follow up on this internally as well and add weight/comments to it.

**Fourth bullet** - I assume this is for the scenario where you have used win+t to bring up the taskbar over the fullscreen window? I believe this is the WinAppSDK implementation that is more "forceful" than (most) other FullScreen implementations in that it won't allow for the minimize action to take effect. If you truly consider this a bug, file an issue instead of a question/discussion for this and we can handle this one in isolation to see if it should be fixed/changed or not.
